### DISCUSSION TITLE
assistance in uploading csv

### DISCUSSION THREAD

**Discussion by cohenanalyticsinc** [influxdata/influxdb#23242] [Q&A]
im having issues uploading csv file (or any data for that matter into the platform) and after following the instructions, i'm at a standstill.  any assistance is greatly appreciated.

**Comment by sanderson** [comment]
@cohenanalyticsinc Can you provide a sample of the CSV you're trying to upload, including the column headers?

**Comment by cohenanalyticsinc** [comment]
I tried 2 csv files

On Wed, Mar 30, 2022 at 12:56 PM Scott Anderson ***@***.***>
wrote:

> @cohenanalyticsinc <https://github.com/cohenanalyticsinc> Can you provide
> a sample of the CSV you're trying to upload, including the column headers?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/influxdata/influxdb/discussions/23242#discussioncomment-2471691>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AYPCVVZYVR4KXGQRBCZL4GDVCSBUXANCNFSM5SAP4KPQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
BJ Cohen
President
Cohen Analytics

**Reply by cohenanalyticsinc** [reply]
[b1.csv](https://github.com/influxdata/influxdb/files/8383303/b1.csv)
[sales.csv](https://github.com/influxdata/influxdb/files/8383305/sales.csv)

**Comment by sanderson** [comment]
To get these CSV files into InfluxDB, you're going to need to use the `influx write` command and inject some additional annotation headers to specify which columns represent the time, fields, and tags. Side note, for this to work, ensure each CSV file uses UTF-8 encoding.

The following commands use the CSV files you provided (updated to be UTF-8 encoded) and writes the data to InfluxDB. Replace `example-bucket` and `path/to/file...` with your actual bucket and file path to each CSV file.

```sh
./influx write -b example-bucket \
  -f path/to/sales.csv \
  --header "#constant measurement,sales" \
  --header "#datatype tag,dateTime:2006-01-02 15:04:05,tag,string"

./influx write -b example-example \
  -f path/to/b1.csv \
  --header "#constant measurement,b1" \
  --header "#datatype tag,tag,tag,tag,double,long,dateTime:1/2/06,tag"
```

This datasets introduce a significant amount of cardinality because of the many unique IDs being written as tags. Consider ignoring columns that don't necessarily provide value in the context of time. Do these IDs actually matter to the type of analysis you're going to be performing on the data. If not, remove them.

**Reply by cohenanalyticsinc** [reply]
thank you - still getting an error message - i copied what you wrote and replaced the file extension and bucket

**Reply by sanderson** [reply]
What error are you getting?

**Reply by cohenanalyticsinc** [reply]
it says failed to write data - invalid line protocol submitted

in order to enter the data - i click on line protocol, enter manually and then use the code:

./influx write -b bjc-bucket \
  -f Users/bcohen/Downloads/b1.csv \
  --header "#constant measurement,b1" \
  --header "#datatype tag,tag,tag,tag,double,long,dateTime:1/2/06,tag"
